projects/cli-files/go-cat/cmd/root.go

```diff
@@ -0,0 +1,108 @@
+package cmd
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+)
+
+func Execute() {
+	// define a flag of type boolean with the specific properties (name, value, usage)
+	numberFlag := flag.Bool("n", false, "number all output lines")
+
+	// parse the command line flags
+	flag.Parse()
+
+	// if there is no argument given then error
+	if len(os.Args) < 2 {
+		fmt.Fprintf(os.Stderr, "no filename provided\n")
+		os.Exit(2)
+	}
+
+	// create a variable to store the file name
+	var filename string
+
+	if *numberFlag {
```

Comment:
If you're using `flag` to parse flags, you probably don't then want to use `os.Args` directly, instead you can use `flag.Args` and `flag.Arg` which will give you "arguments left over after all the flags have been parsed", avoiding the need to compensate for them yourself :)

Reply:
Have adjusted this to only use the flags package
And noted flags.Parse() and flags.Args() and what each does